api/crl.go

```diff
@@ -0,0 +1,52 @@
+package api
+
+import (
+	"encoding/pem"
+	"fmt"
+	"github.com/pkg/errors"
+	"net/http"
+)
+
+// CRL is an HTTP handler that returns the current CRL in DER or PEM format
+func (h *caHandler) CRL(w http.ResponseWriter, r *http.Request) {
+	crlBytes, err := h.Authority.GetCertificateRevocationList()
+
+	_, formatAsPEM := r.URL.Query()["pem"]
+
+	if err != nil {
+		w.WriteHeader(500)
+		_, err = fmt.Fprintf(w, "%v\n", err)
+		if err != nil {
+			panic(errors.Wrap(err, "error writing http response"))
+		}
+		return
+	}
+
+	if crlBytes == nil {
+		w.WriteHeader(404)
+		_, err = fmt.Fprintln(w, "No CRL available")
+		if err != nil {
+			panic(errors.Wrap(err, "error writing http response"))
+		}
+		return
+	}
```

Comment:
Use [`http.NotFound(w, r)`](https://pkg.go.dev/net/http#NotFound) here

Reply:
Well in fact, `GetCertificateRevocationList` returns a 404 if CRLs are not enabled, we can detect that error and use `http.NotFound`.

Reply:
I used the generic http.Error() instead so its a bit neater